OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -236,15 +238,39 @@ void ITick.Tick(Actor self)
 
 		protected virtual void Tick(Actor self)
 		{
-			while (queue.Count > 0 && BuildableItems().All(b => b.Name != queue[0].Item))
+			foreach (var item in queue.ToArray())
```

Comment:
you should not need to do a ToArray here and you could probably use a select with this as a predicate (BuildableItems().All(b => b.Name != item.Item)).